### PR TITLE
fix: dont show generate button if create permission for action and pages is absent

### LINKED ISSUE #26181: Handle generate page functionality in EE for RBAC
Generate Page button on Google Sheet Schema page should be behind RBAC controls. If the user doesn't have permissions : 
- to create page
- to create action

should not be able to see generate page button

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description
- Removed the display of generate button on google sheet datasource page for schema preview if the user lacks the permissions for creating actions or pages. Added checks for `create:pageActions` and `create:pages` permission to show generate page button. 
- Disabled the `generate new page` on datasource cards on active datasources page

#### PR fixes following issue(s)
Fixes #26181
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- Chore (housekeeping or task changes that don't impact user perception)
- This change requires a documentation update
>
>
>
- [x] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [x] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/pages/Editor/SaaSEditor/GoogleSheetSchema.tsx [hunk 1]
```diff
@@ -268,6 +277,21 @@ function GoogleSheetSchema(props: Props) {
...
 
+  const pagePermissions = useSelector(getPagePermissions);
+  const canCreateActions = hasCreateActionPermission(pagePermissions);
+
+  const userAppPermissions = useSelector(
+    (state: AppState) => getCurrentApplication(state)?.userPermissions ?? [],
+  );
+  const canCreatePages = hasCreatePagePermission(userAppPermissions);
+
+  const showSubmitButton =
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/SaaSEditor/GoogleSheetSchema.tsx:288]: can we change the variable name to `showGenneratePageBtn`?

File: app/client/src/pages/Editor/IntegrationEditor/DatasourceCard.tsx [hunk 2]
```diff
@@ -300,8 +312,9 @@ function DatasourceCard(props: DatasourceCardProps) {
             {supportTemplateGeneration && !showReconnectButton && (
               <Button
                 className={"t--generate-template"}
+                disable={canGeneratePage}
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/IntegrationEditor/DatasourceCard.tsx:315]: I think this should be `isDisabled`

**Review by ankitakinger — DISMISSED** [review]
LGTM

**Review by ankitakinger — APPROVED** [review]
LGTM

**Review by ayushpahwa — APPROVED** [review]

**Pull request merged**